tests/test_structured_document.py

```diff
@@ -57,6 +49,205 @@ def test_manifestation_date(self):
         assert_equal(datestring(a.manifestation_date), '2012-01-02')
         assert_is_instance(a.manifestation_date, date)
 
+    def test_language(self):
+        a = Act()
+        a.language = 'fre'
+        assert_equal(a.language, 'fre')
+
+    def test_namespaces(self):
+        # default for the time being is still AKN2
+        a = Act()
+        assert_equal(a.namespace, 'http://www.akomantoso.org/2.0')
+
+        # prefer AKN3 when both 2 and 3 are listed as namespaces
+        a = Act(xml="""<?xml version="1.0"?>
+<foo:akomaNtoso xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:foo="http://www.akomantoso.org/2.0" xmlns:bar="http://docs.oasis-open.org/legaldocml/ns/akn/3.0" xsi:schemaLocation="http://www.akomantoso.org/2.0 akomantoso20.xsd">
+  <foo:act>
+    <meta/>
+    <foo:body/>
+  </foo:act>
+</foo:akomaNtoso>""")
+        assert_equal(a.namespace, 'http://docs.oasis-open.org/legaldocml/ns/akn/3.0')
+
+        # prefer AKN2 when 2 and something else are listed as namespaces
+        a = Act(xml="""<?xml version="1.0"?>
+<foo:akomaNtoso xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:foo="http://www.akomantoso.org/2.0" xmlns:bar="http://docs.oasis-open.org/legaldocml/ns/akn/5.0" xsi:schemaLocation="http://www.akomantoso.org/2.0 akomantoso20.xsd">
+  <foo:act>
+    <meta/>
+    <foo:body/>
+  </foo:act>
+</foo:akomaNtoso>""")
+        assert_equal(a.namespace, 'http://www.akomantoso.org/2.0')
+
+        # throw error if neither of AKN2 and AKN3 are listed as namespaces
+        with assert_raises(ValueError) as raised:
+            Act(xml="""<?xml version="1.0"?>
+                <foo:akomaNtoso xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:foo="http://www.akomantoso.org/4.0" xmlns:bar="http://docs.oasis-open.org/legaldocml/ns/akn/5.0" xsi:schemaLocation="http://www.akomantoso.org/2.0 akomantoso20.xsd">
+  <foo:act>
+    <meta/>
+    <body/>
+  </foo:act>
+</foo:akomaNtoso>""")
+        assert_in(
+            "Expected to find one of the following Akoma Ntoso XML namespaces: http://docs.oasis-open.org/legaldocml/ns/akn/3.0, http://www.akomantoso.org/2.0. Only these namespaces were found: http://www.w3.org/2001/XMLSchema-instance, http://www.akomantoso.org/4.0, http://docs.oasis-open.org/legaldocml/ns/akn/5.0",
+            raised.exception.args)
+
+        # ignore content that isn't in the chosen namespace
+        a = Act(xml="""<?xml version="1.0"?>
+<bar:akomaNtoso xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:foo="http://www.akomantoso.org/2.0" xmlns:bar="http://docs.oasis-open.org/legaldocml/ns/akn/3.0" xsi:schemaLocation="http://www.akomantoso.org/2.0 akomantoso20.xsd">
+  <bar:act>
+    <bar:meta/>
+    <foo:body>
+      <section id="section-1">
+        <content>
+          <p>This content should be ignored as it's using the AKN2 namespace.</p>
```

Comment:
if objectify does ignore namespaces and just chooses the first matching element, then to get it to fail we'd want our incorrect element to show up just before the real one. So I'd put the bad `<bar:p>` element just before line 103 and remove `bar:body` completely.